test/extended/networking/ipsec.go

```diff
@@ -0,0 +1,709 @@
+		ipsecRolloutWaitDuration, true, func(ctx context.Context) (bool, error) {
+			done, err := isIPsecDaemonSetRunning(oc)
+			if err != nil && !isConnResetErr(err) {
+				return false, err
+			}
+			if done {
+				done, err = isClusterOperatorsReady((oc))
+				if err != nil && !isConnResetErr(err) {
+					return false, err
+				}
+			}
+			return done, nil
+		})
+}
+
+// ensureIPsecMachineConfigRolloutComplete this function ensures ipsec machine config extension is rolled out
+// on all of master and worked nodes and cluster operators are coming back into ready state
+// once ipsec rollout is complete.
+func ensureIPsecMachineConfigRolloutComplete(oc *exutil.CLI) error {
+	return wait.PollUntilContextTimeout(context.Background(), ipsecRolloutWaitInterval,
+		ipsecRolloutWaitDuration, true, func(ctx context.Context) (bool, error) {
+			done, err := isMachineConfigPoolReadyWithIPsec(oc)
+			if err != nil && !isConnResetErr(err) {
+				return false, err
+			}
+			if done {
+				done, err = isClusterOperatorsReady((oc))
+				if err != nil && !isConnResetErr(err) {
+					return false, err
+				}
+			}
+			return done, nil
+		})
+}
+
+// ensureIPsecDisabled this function ensure IPsec is disabled by making sure ovn-ipsec-host daemonset
+// is completely removed from the cluster and cluster operators are coming back into ready state
+// once ipsec rollout is complete.
+func ensureIPsecDisabled(oc *exutil.CLI) error {
+	return wait.PollUntilContextTimeout(context.Background(), ipsecRolloutWaitInterval,
+		ipsecRolloutWaitDuration, true, func(ctx context.Context) (bool, error) {
+			ds, err := getIPsecDaemonSet(oc)
+			if err != nil && !isConnResetErr(err) {
+				return false, err
+			}
+			var done bool
+			if ds == nil && err == nil {
+				done, err = isClusterOperatorsReady((oc))
+				if err != nil && !isConnResetErr(err) {
+					return false, err
+				}
+			}
+			return done, nil
+		})
+}
+
+// This checks master and worker machine config pool status are set with ipsec
+// extension which confirms extension is successfully rolled out on all nodes.
+func isMachineConfigPoolReadyWithIPsec(oc *exutil.CLI) (bool, error) {
+	masterWithIPsec, err := isMasterMachineConfigPoolsWithIPsec(oc)
+	if err != nil {
+		return false, err
+	}
+	workerWithIPsec, err := isWorkerMachineConfigPoolsReadyWithIPsec(oc)
+	if err != nil {
+		return false, err
+	}
+	return masterWithIPsec && workerWithIPsec, nil
+}
+
+func isMasterMachineConfigPoolsWithIPsec(oc *exutil.CLI) (bool, error) {
+	masterLabelSelector := labels.SelectorFromSet(labels.Set{"machineconfiguration.openshift.io/role": "master"})
+	return isMachineConfigPoolsReady(oc, masterLabelSelector, masterIPsecMachineConfigName, true)
+}
+
+func isWorkerMachineConfigPoolsReadyWithIPsec(oc *exutil.CLI) (bool, error) {
+	return isWorkerMachineConfigPoolsReady(oc, workerIPSecMachineConfigName, true)
+}
+
+func isWorkerMachineConfigPoolsReady(oc *exutil.CLI, machineConfigName string, mustExist bool) (bool, error) {
```

Comment:
nit: some functions with general purpose could be moved to util.go: `isWorkerMachineConfigPoolsReady`, `isMachineConfigPoolsReady`, `isMachineConfigReadyInPool`, `isMachineConfigPoolReadyWithoutMachineConfig`, `hasSourceInMachineConfigStatus`, `isClusterOperatorsReady`

Also using `are` instead of `is` verb sound more fitting for some of this with plural subject.